tensorflow/compiler/tf2tensorrt/segment/segment.cc

```diff
@@ -681,13 +682,80 @@ void AddSegmentForNode(const grappler::GraphProperties* graph_properties,
 
 }  // namespace
 
+Status ExportNonConversionReportToCSV(
+  string filename,
+  std::map<string, std::map<string, int>>& nonconverted_ops_map) {
+  std::fstream csv_file(filename, std::fstream::out | std::fstream::trunc);
+
+  LOG(WARNING) << "TF-TRT Non-Conversion Report saved at: `" << filename << "`";
+
+  if (!csv_file) {
+      return errors::Internal("Failed to open output file: `", filename, "`");
+  }
+
+  string sep = "|";
+
+  csv_file << "OP Name" << sep << "Reason" << sep << "Count" << std::endl;
+
+  for (auto& op_details : nonconverted_ops_map) {
+      auto op_name = op_details.first;
+      auto op_data = op_details.second;
+
+      for (auto& reject_data : op_data) {
+          auto reason = reject_data.first;
+          auto count = reject_data.second;
+          csv_file << op_name << sep << reason << sep << count << std::endl;
+      }
+  }
+
+  if (!csv_file.bad() && !csv_file.fail()) {
+    return Status::OK();
+  } else {
+    return errors::Internal(
+      "Error closing the file `", filename, "`. "
+      "The file might be corrupted."
+    );
+  }
+}
+
 string GenerateNonConversionReport(
     std::map<string, std::map<string, int>>& nonconverted_ops_map) {
   // Fetch whether to print a detailed version of the TF-TRT conversion report.
-  bool show_detailed_conversion_report;
-  TF_CHECK_OK(ReadBoolFromEnvVar("TF_TRT_SHOW_DETAILED_REPORT",
-                                 /*default_value=*/false,
-                                 &show_detailed_conversion_report));
+  // TF_TRT_SHOW_DETAILED_REPORT triggers three possible behaviors:
+  // - If Number >= 1:      Print detailed non-conversion report on stdout.
+  //                        Usage: TF_TRT_SHOW_DETAILED_REPORT=1
+  // - If non empty string: Exports the non-conversion report in CSV format at
+  //                        the path defined by the environment variable.
+  //                        This will also print the detailed non-conversion
+  //                        report on stdout.
+  //                        Usage: TF_TRT_SHOW_DETAILED_REPORT=/path/to/file.csv
+  // - Else:                Print normal (undetailed) non-conversion report on
+  //                        stdout.
```

Comment:
Shall we make this more restrict like this:
0: no report
1: report
everything else: treat as a path for the output filename

Reply:
it's safer to follow the approach I took because:
- the default case is `else` and represents the normal report => not an output filename
- If the number is >= 1 print a detailed report (useful to be >= 1 and not == 1 because it could allow some verbosity control in the future)
- Non empty str => it's a filepath


I think this is the most flexible and future proof approach